miio/viomidishwasher.py

```diff
@@ -0,0 +1,529 @@
+import enum
+import logging
+from collections import defaultdict
+from datetime import datetime, timedelta
+from typing import Any, Dict, List, Optional
+
+import click
+
+from .click_common import EnumType, command, format_output
+from .device import Device
+from .exceptions import DeviceException
+
+_LOGGER = logging.getLogger(__name__)
+
+MODEL_DISWAHSER_M02 = "viomi.dishwasher.m02"
+
+MODELS_SUPPORTED = [MODEL_DISWAHSER_M02]
+
+
+class MachineStatus(enum.IntEnum):
+    Off = 0
+    On = 1
+    Running = 2
+    Paused = 3
+    Scheduled = 5
+
+
+class ProgramStatus(enum.IntEnum):
+    Standby = 0
+    Prewash = 1
+    Wash = 2
+    Rinse = 3
+    Drying = 4
+    Unknown = -1
+
+
+class Program(enum.IntEnum):
+    Standard = 0
+    Eco = 1
+    Quick = 2
+    Intensive = 3
+    Glassware = 4
+    Sterilize = 7
+    Unknown = -1
+
+    @property
+    def run_time(self):
+        return ProgramRunTime[self.value]
+
+
+ProgramRunTime = {
+    Program.Standard: 7102,
+    Program.Eco: 7702,
+    Program.Quick: 1675,
+    Program.Intensive: 7522,
+    Program.Glassware: 6930,
+    Program.Sterilize: 8295,
+    Program.Unknown: -1,
+}
+
+
+class ChildLockStatus(enum.IntEnum):
+    Enabled = 1
+    Disabled = 0
+
+
+class DoorStatus(enum.IntEnum):
+    Open = 128
+    Closed = 0
+
+
+class SystemStatus(enum.IntEnum):
+    WaterLeak = 1
+    InsufficientWaterFlow = 4
+    InternalConnectionError = 9
+    ThermistorError = 32
+    InsufficientWaterSoftener = 512
+    HeatingElementError = 2048
+
+
+class ViomiDishwasherStatus:
+    def __init__(self, data: Dict[str, int]) -> None:
+        """A ViomiDishwasherStatus representing the most important values for the device.
+
+        Example:
+            Example payload
+                {
+                    "child_lock": 0,
+                    "program": 2,
+                    "run_status": 512,
+                    "wash_status": 0,
+                    "wash_temp": 86,
+                    "power": 0,
+                    "left_time": 0,
+                    "wash_done_appointment": 0,
+                    "freshdry_interval": 0,
+                    "wash_process": 0
+                }
```

Comment:
```suggestion
            {
                "child_lock": 0,
                "program": 2,
                "run_status": 512,
                "wash_status": 0,
                "wash_temp": 86,
                "power": 0,
                "left_time": 0,
                "wash_done_appointment": 0,
                "freshdry_interval": 0,
                "wash_process": 0
            }
```

Reply:
`Example:` is already enough to indicate that the following data is an example of what to expect.